synapse/rest/client/account.py

```diff
@@ -79,43 +83,29 @@ async def on_POST(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
                     "User password resets have been disabled due to lack of email config"
                 )
             raise SynapseError(
-                400, "Email-based password resets have been disabled on this server"
+                HTTPStatus.NOT_FOUND,
```

Comment:
Is there a reason why this is changed from `bad request` to `not found`?

Reply:
The request in an of itself is perfectly fine; we've just chosen not to serve it. Is there a better code choice? 403 Forbidden maybe?